docs/language_guide.md

```diff
@@ -659,18 +705,19 @@ Wallet :: (owner) {
 
 Flint introduces the concept of **caller capabilities**. While traditional computer programs have an entry point (the _main_ function), smart contracts do not. After a contract is deployed on the blockchain, its code does not run until an Ethereum transaction is received. Smart contracts are in fact more akin to RESTful web services presenting API endpoints. It is important to prevent unauthorized parties from calling sensitive functions.
 
-In Flint, functions of a contract are declared within caller capability blocks, which restrict which users/contracts are allowed to call the enclosed functions.
-```swift
-// Only the manager of the Bank can call "clear".
-Bank :: (manager) { // manager is a state property.
-  func clear(address: Address) {
-    // body
-  }
-}
-```
-Caller capabilities can be any property declared in the contract's declaration, as long as their type is `Address` or `[Address]`.
+Solidity uses function modifiers to insert dynamic checks in functions, which can for instance abort unauthorised calls. However, it is easy to forget to specify these checks, as the language does not require programmers to write them.
+Having a language construct which protects functions from invalid calls could require programmers to systematically think about which parties should be able to call the functions they are about to define.
+
+In Flint, functions of a contract are declared within caller capability blocks, which protect the functions from invalid access.
 
-Note: The special caller capability any allows `any` caller to execute the function in the group.
+### Caller Group
+They contain a caller group, namely `(any)`, `(child)` and `(child, parent)`. The transaction caller needs to hold _any_ of the addresses contained within the caller group which is a a union of its members.
```

Comment:
This makes no sense. Perhaps something like this would be better?

> Caller Groups consist of a list of either state properties holding `Address`es or the `any` keyword, enclosed in parentheses. For example, `(any)`, `(administrator)`, and `(owner, managers)` are all valid Caller Groups. Functions inside these caller groups can only be called by an Ethereum address that is a member of at least one of the state properties, except when `any` has been used. In the case of `(any)`, there are no restrictions on the address of the caller.

Reply:
Not technically true - it's not just a list of state properties or the any keyword. Which is part of the reason why it's unintelligible because I was trying to stay more generic. I'll have a go at rewriting.